### PR TITLE
Add node-api-stub

### PR DESCRIPTION
**Pull request by Freed-Wu**
* Before adding new features and new modules, please go to issues to submit the relevant feature description first.
* Write good commit messages and use the same coding conventions as the rest of the project.
* Please commit code to dev branch and we will merge into master branch in feature
* Ensure your edited codes with four spaces instead of TAB.

------

* 增加新特性和新模块之前，请先到issues提交相关特性说明，经过讨论评估确认后，再进行相应的代码提交，避免做无用工作。
* 编写友好可读的提交信息，并使用与工程代码相同的代码规范，代码请用4个空格字符代替tab缩进。
* 请提交代码到dev分支，如果通过，我们会在特定时间合并到master分支上。
* 为了规范化提交日志的格式，commit消息，不要用中文，请用英文描述。

### CODE REVIEW

File: packages/n/node-api-stub/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    add_urls("https://github.com/napi-bindings/node-api-stub.git")
+    add_urls("https://github.com/napi-bindings/node-api-stub/archive/refs/tags/$(version).tar.gz")
+    add_versions("8.0.0", "7fdf725a3122f4d86443e707227a30c663c7163c6d1b9f883cb2305f18e01740")
+    add_patches("8.0.0", path.join(os.scriptdir(), "patches", "cmake.patch"), "ed3edac79b7efa682553e6dc85b5479f111445fc4df2e1b9674c5d7583cfb945")
+
+    add_deps("cmake")
+    add_links("node_api")
+    on_install("windows", "macosx", "linux", function (package)
+        import("package.tools.cmake").install(package)
+    end)
```
Comment (waruqi) [inline packages/n/node-api-stub/xmake.lua:17]: add on_test

File: packages/n/node-api-stub/xmake.lua [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package("node-api-stub")
+
+    set_kind("static")
```
Comment (waruqi) [inline packages/n/node-api-stub/xmake.lua:3]: no this kind, remove it

File: packages/n/node-api-stub/xmake.lua [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+    add_urls("https://github.com/napi-bindings/node-api-stub.git")
+    add_urls("https://github.com/napi-bindings/node-api-stub/archive/refs/tags/$(version).tar.gz")
+    add_versions("8.0.0", "7fdf725a3122f4d86443e707227a30c663c7163c6d1b9f883cb2305f18e01740")
+    add_patches("8.0.0", path.join(os.scriptdir(), "patches", "cmake.patch"), "ed3edac79b7efa682553e6dc85b5479f111445fc4df2e1b9674c5d7583cfb945")
+
+    add_deps("cmake")
+    add_links("node_api")
+    on_install("windows", "macosx", "linux", function (package)
+        import("package.tools.cmake").install(package)
```
Comment (waruqi) [inline packages/n/node-api-stub/xmake.lua:16]: please add shared/static configs

Comment (Freed-Wu) [reply packages/n/node-api-stub/xmake.lua:16]: https://github.com/napi-bindings/node-api-stub only have static config

File: packages/n/node-api-stub/xmake.lua (hunk 1 above)
Comment (Freed-Wu) [reply packages/n/node-api-stub/xmake.lua:17]: This package only provide a static library, even the header files is provide by node-api-headers. We cannot check it standonly.

Comment (waruqi) [reply packages/n/node-api-stub/xmake.lua:17]: on_test is required

Comment (Freed-Wu) [reply packages/n/node-api-stub/xmake.lua:17]: OK

File: packages/n/node-api-stub/xmake.lua (hunk 3 above)
Comment (waruqi) [reply packages/n/node-api-stub/xmake.lua:16]: you need set shared config as readonly. 


https://github.com/xmake-io/xmake-repo/blob/6db95290eef9c4acd4c3d9b3369d074de70868a5/packages/v/vc/xmake.lua#L10

File: packages/n/node-api-stub/xmake.lua [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+    add_deps("cmake")
+    add_links("node_api")
+
+    on_install("windows", "macosx", "linux", function (package)
+        import("package.tools.cmake").install(package)
+    end)
+
+    on_test(function (package)
+        -- just check if function napi_get_undefined() exists
```
Comment (waruqi) [inline packages/n/node-api-stub/xmake.lua:20]: remove comments

File: packages/n/node-api-stub/xmake.lua [hunk 5]
```diff
@@ -0,0 +1,27 @@
+package("node-api-stub")
+
+    set_homepage("https://github.com/napi-bindings/node-api-stub")
+    set_description("Node-API stub")
+    set_license("MIT")
+
+    add_urls("https://github.com/napi-bindings/node-api-stub.git")
+    add_urls("https://github.com/napi-bindings/node-api-stub/archive/refs/tags/$(version).tar.gz")
```
Comment (waruqi) [inline packages/n/node-api-stub/xmake.lua:8]: add newline

**Comment by Freed-Wu** [comment]
A test has been added.

**Comment by waruqi** [comment]
please fix ci test errors

**Comment by Freed-Wu** [comment]
why `add_links("node_api")` don't make the result `gcc -lnode_api`?

**Comment by waruqi** [comment]
> why `add_links("node_api")` don't make the result `gcc -lnode_api`?

because it is a compilation error. No test link yet

> > checking for c links(node_api)
> checking for c snippet(test)
checkinfo: ...amdir/core/sandbox/modules/import/core/tool/compiler.lua:84: @programdir/modules/core/tools/gcc.lua:966: /var/folders/9f/9p4dh6hs5yddrk7drxq8rc_80000gn/T/.xmake501/241219/_A03737E1A4914FBF9454B8AA702B5626.c:6:17: error: call to undeclared function 'napi_get_undefined'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
                napi_get_undefined(p, q);
                ^

**Pull request merged**